src/libs/libpng/lv_libpng.c

```diff
@@ -109,6 +111,33 @@ static lv_result_t decoder_info(lv_image_decoder_t * decoder, lv_image_decoder_d
 
         return LV_RESULT_OK;
     }
+    /*If it's a PNG file in a  C array...*/
+    else if(src_type == LV_IMAGE_SRC_VARIABLE) {
+        const lv_image_dsc_t * img_dsc = src;
+        const uint32_t data_size = img_dsc->data_size;
+        size = ((uint32_t *)img_dsc->data) + 4;
+
+        if(data_size < sizeof(magic)) return LV_RESULT_INVALID;
+        if(lv_memcmp(magic, img_dsc->data, sizeof(magic))) return LV_RESULT_INVALID;
+
+        header->cf = LV_COLOR_FORMAT_ARGB8888;
+
+        if(img_dsc->header.w) {
+            header->w = img_dsc->header.w;         /*Save the image width*/
+        }
+        else {
+            header->w = (int32_t)((size[0] & 0xff000000) >> 24) + ((size[0] & 0x00ff0000) >> 8);
+        }
+
+        if(img_dsc->header.h) {
+            header->h = img_dsc->header.h;         /*Save the color height*/
+        }
+        else {
+            header->h = (int32_t)((size[1] & 0xff000000) >> 24) + ((size[1] & 0x00ff0000) >> 8);
+        }
```

Comment:
Can we re-use the same logic for file type of image?

Reply:
I merged the file and VARIABLE. I applied the same modification method to lodepng.